pkg/graveler/graveler.go

```diff
@@ -1335,14 +1335,17 @@ func (g *KVGraveler) Get(ctx context.Context, repositoryID RepositoryID, ref Ref
 	if reference.StagingToken != "" {
 		// try to get from staging, if not found proceed to committed
 		value, err := g.getFromStagingArea(ctx, reference.Branch, key)
-		if err != nil && !errors.Is(err, ErrNotFound) {
-			return nil, err
-		}
-		if value == nil {
-			// tombstone - the entry was deleted on the branch => doesn't exist
-			return nil, ErrNotFound
+		if err != nil {
+			if !errors.Is(err, ErrNotFound) {
+				return nil, err
+			} // proceed to committed
+		} else { // In staging
+			if value == nil {
+				// tombstone - the entry was deleted on the branch => doesn't exist
+				return nil, ErrNotFound
+			}
+			return value, nil
```

Comment:
Great catch!!

I think it deserves its own PR. Also, there are tests for `Get` flow in Graveler. Can we add/change them to fit this scenario too?

Reply:
The tests were there, they were just outdated. Unfortunately I suspect we have many such tests where the flow is not tested correctly due to mock configurations. We should take action once we remove the DB support to refactor the tests and clean up the mocks